nipype/pipeline/engine/nodes.py

```diff
@@ -295,7 +295,8 @@ def output_dir(self):
         if self.parameterization:
             params_str = ["{}".format(p) for p in self.parameterization]
             if not str2bool(self.config["execution"]["parameterize_dirs"]):
-                params_str = [_parameterization_dir(p) for p in params_str]
+                params_str = [_parameterization_dir(p,32) for p in params_str]
+            params_str = [_parameterization_dir(p,252) for p in params_str]
```

Comment:
Feels a little weird to call it twice, with the second one doing nothing. But this whole block could be simplified:

```Python
        if self.parameterization:
            maxlen = 252 if str2bool(self.config["execution"]["parameterize_dirs"]) else 32
            params_str = [_parameterization_dir(str(p), maxlen) for p in self.parameterization]
            outputdir = op.join(outputdir, *params_str)
```

Reply:
I'm not a Python programmer but that seems fine too. The only functional different I see is that you are now also transforming your components through `str(p)` . I am not sure what that does but maybe it handles UTF-8 encoded strings better? (A directory component's length is limited by bytes, not characters, so we also have to be careful there).

Fee free to apply this change.

Reply:
Or I can make the change myself too if you approve (your own code!).

At the same time, I'll probably set 32 as the default value for the maxlen argument in the other file.

Reply:
Okay, I'll make the changes.